nomad/state/state_store.go

```diff
@@ -1894,6 +1894,68 @@ func (s *StateStore) DeleteJobTxn(index uint64, namespace, jobID string, txn Txn
 		return err
 	}
 
+	// Delete job deployments
+	deployments, err := s.DeploymentsByJobID(nil, namespace, job.ID, true)
+	if err != nil {
+		return fmt.Errorf("deployment lookup for job %s failed: %v", job.ID, err)
+	}
+
+	for _, deployment := range deployments {
+		// Delete the deployment
+		if err := txn.Delete("deployment", deployment); err != nil {
+			return fmt.Errorf("deployment delete failed: %v", err)
+		}
+		if err := txn.Insert("index", &IndexEntry{"deployment", index}); err != nil {
+			return fmt.Errorf("index update failed: %v", err)
+		}
+	}
+
+	// Mark all "pending" evals for this job as "complete"
+	evals, err := s.EvalsByJob(nil, namespace, job.ID)
+	if err != nil {
+		return fmt.Errorf("eval lookup for job %s failed: %v", job.ID, err)
+	}
+
+	for _, eval := range evals {
+		existing, err := txn.First("evals", "id", eval.ID)
+		if err != nil {
+			return fmt.Errorf("eval lookup failed: %v", err)
+		}
+		if existing == nil {
+			continue
+		}
+
+		eval := existing.(*structs.Evaluation).Copy()
+		if eval.Status != structs.EvalStatusPending {
+			continue
+		}
+		eval.Status = structs.EvalStatusComplete
+		eval.StatusDescription = fmt.Sprintf("evaluation deleted while purging job %s", job.ID)
```

Comment:
```suggestion
		eval.StatusDescription = fmt.Sprintf("job %s deleted", job.ID)
```

The eval has not been deleted.